drivers/pwm/pwm_nrfx.c

```diff
@@ -357,7 +362,8 @@ static int pwm_nrfx_pm_action(const struct device *dev,
 			.load_mode = NRF_PWM_LOAD_INDIVIDUAL,		      \
 			.step_mode = NRF_PWM_STEP_TRIGGERED,		      \
 		},							      \
-		.seq.values.p_raw = pwm_nrfx_##idx##_data.seq_values,	      \
+		.seq_values = pwm_##idx##_seq_values,			      \
+		.seq.values.p_raw = pwm_##idx##_seq_values,		      \
```

Comment:
The `seq_values` pointer is redundant. You can directly use `seq.values.p_raw` instead. I did not realize this earlier.

Reply:
I have tried that before and unfortunately you can't, because this is defined as `uint16_t const *` pointer: https://github.com/zephyrproject-rtos/hal_nordic/blob/bdef8b66d5f59d95c09889918a04ddaecce322c8/nrfx/hal/nrf_pwm.h#L252

I guess this `const` qualifier is used there to mark that the PWM HAL/driver APIs do not modify the contents of PWM data buffer internally

Reply:
Right. I went a bit too far with this "directly". But it's just a pointer and it is assigned with a modifiable array in this shim. I think it's justified to just typecast it when the array needs to be modified (maybe with an inline function or a macro so that it does not look too hacky).

Reply:
done, please take a look

Reply:
Nice. Perhaps it would be better to name the function `seq_values_ptr_get()` so that it corresponds with the name of the actual array.

Reply:
done